packages/esm-appointments-app/src/appointment-forms/create-appointment-form.component.tsx

```diff
@@ -0,0 +1,383 @@
+      return;
+    }
+
+    const service = services.find((service) => service.name === selectedService);
+
+    const serviceUuid = services.find((service) => service.name === selectedService)?.uuid;
+
+    const [startHours, startMinutes] = convertTime12to24(startTime, timeFormat);
+    const [endHours, endMinutes] = convertTime12to24(startTime, timeFormat);
+
+    const startDateTime = new Date(
+      dayjs(startDate).year(),
+      dayjs(startDate).month(),
+      dayjs(startDate).date(),
+      startHours,
+      startMinutes,
+    );
+
+    const endDateTime = new Date(
+      dayjs(startDate).year(),
+      dayjs(startDate).month(),
+      dayjs(startDate).date(),
+      endHours,
+      endMinutes,
+    );
+
+    const appointmentPayload: AppointmentPayload = {
+      appointmentKind,
+      serviceUuid,
+      startDateTime: dayjs(startDateTime).format(),
+      endDateTime: dayjs(endDateTime).format(),
+      providerUuid: selectedProvider,
+      locationUuid: userLocation,
+      patientUuid: patientUuid,
+      providers: [],
+      comments: appointmentNote,
+    };
+
+    const abortController = new AbortController();
+    createAppointment(appointmentPayload, abortController).then(
+      ({ status }) => {
+        if (status === 200) {
+          closeOverlay();
+          fetchAppointments(abortController);
+          showToast({
+            critical: true,
+            kind: 'success',
+            description: t('appointmentNowVisible', 'It is now visible on the Appointments page'),
+            title: t('appointmentScheduled', 'Appointment scheduled'),
+          });
+        }
+
+        mutate(`${appointmentsSearchUrl}`);
+      },
+      (error) => {
+        showNotification({
+          title: t('appointmentFormError', 'Error scheduling appointment'),
+          kind: 'error',
+          critical: true,
+          description: error?.message,
+        });
+      },
+    );
+  };
+
+  if (isLoading) {
+    return <SearchSkeleton role="progressbar" />;
+  }
+  return (
+    <div>
+      {patient ? (
+        <div className={styles.patientInfo}>
+          <ExtensionSlot
+            extensionSlotName="patient-header-slot"
+            state={{
+              patient,
+              patientUuid: patient.id,
+            }}
+          />
+        </div>
+      ) : null}
+      <div className={styles.formWrapper}>
+        <section className={styles.formGroup}>
+          <span>{t('appointmentDateTime', 'Appointment Date and Time')}</span>
+          <div className={styles.dateTimeContainer}>
+            <Toggle
+              className={styles.toggle}
+              aria-label="toggle button"
+              id="allDay"
+              toggled={allDayOn}
+              labelText="All Day"
+              onToggle={() => setAllDayOn(!allDayOn)}
+            />
+            <div className={styles.datePickerControl}>
+              <DatePicker
+                datePickerType="single"
+                dateFormat="d/m/Y"
```

Comment:
This time format should be implementation specific

Reply:
I don't understand this comment @ZacButko

Reply:
Different countries will format a date like d/m/y while others use m/d/y and still others use y/m/d. It should be enabled for an implementation to specify their preferred date format